new-lamassu-admin/src/components/NotificationCenter/NotificationCenter.js

```diff
@@ -0,0 +1,143 @@
+    }
+    hasUnreadNotifications
+    machines {
+      deviceId
+      name
+    }
+  }
+`
+
+const CLEAR_NOTIFICATION = gql`
+  mutation clearNotification($id: ID!) {
+    clearNotification(id: $id) {
+      id
+    }
+  }
+`
+
+const CLEAR_ALL_NOTIFICATIONS = gql`
+  mutation clearAllNotifications {
+    clearAllNotifications {
+      id
+    }
+  }
+`
+
+const NotificationCenter = ({ close, notifyUnread }) => {
+  const classes = useStyles()
+  const { data, loading } = useQuery(GET_NOTIFICATIONS)
+  const [showingUnread, setShowingUnread] = useState(false)
+  const machines = R.compose(
+    R.map(R.prop('name')),
+    R.indexBy(R.prop('deviceId'))
+  )(data?.machines ?? [])
+
+  const notifications = R.path(['notifications'])(data) ?? []
+  const hasUnread = data?.hasUnreadNotifications ?? false
+  if (!hasUnread) {
+    notifyUnread && notifyUnread()
+  }
+  const [clearNotification] = useMutation(CLEAR_NOTIFICATION, {
+    onError: () => console.error('Error while clearing notification'),
+    refetchQueries: () => ['getNotifications']
+  })
+  const [clearAllNotifications] = useMutation(CLEAR_ALL_NOTIFICATIONS, {
+    onError: () => console.error('Error while clearing all notifications'),
+    refetchQueries: () => ['getNotifications']
+  })
+
+  const handleClearNotification = id => {
+    clearNotification({ variables: { id } })
+  }
+  const buildNotifications = () => {
+    const notificationsToShow =
+      !showingUnread || !hasUnread
+        ? notifications
+        : R.filter(R.propEq('read', false))(notifications)
+    return notificationsToShow.map(n => {
+      return (
+        <NotificationRow
+          key={n.id}
+          id={n.id}
+          type={n.type}
+          detail={n.detail}
+          message={n.message}
+          deviceName={machines[n.detail.deviceId]}
+          created={n.created}
+          read={n.read}
+          valid={n.valid}
+          onClear={handleClearNotification}
+        />
+      )
+    })
+  }
+
+  return (
+    <>
+      <div className={classes.container}>
+        <div className={classes.header}>
+          <H5 className={classes.headerText}>Notifications</H5>
+          <button onClick={close} className={classes.notificationIcon}>
+            <NotificationIconZodiac />
+            {hasUnread && <div className={classes.hasUnread} />}
+          </button>
+        </div>
+        <div className={classes.actionButtons}>
+          {hasUnread && (
+            <ActionButton
+              color="primary"
+              Icon={ShowUnreadIcon}
+              InverseIcon={ClearAllIconInverse}
+              style={{ ...styles.clearAllButton }}
+              onClick={() => setShowingUnread(!showingUnread)}>
+              {showingUnread ? 'Show all' : 'Show unread'}
+            </ActionButton>
+          )}
+          <ActionButton
+            color="primary"
+            Icon={ClearAllIcon}
+            InverseIcon={ClearAllIconInverse}
+            style={{ ...styles.clearAllButton }}
```

Comment:
To avoid inline styling and keep it all in the stylesheet, check the changes I made to the ActionButton container on another PR

https://github.com/lamassu/lamassu-server/pull/537/files#diff-38b5504c808007220c5af0ac5dcab3170817148adcf9966a8605527f08dfcb74